utility/src/test/java/org/oppia/android/util/parser/html/PolicyPageTagHandlerTest.kt

```diff
@@ -0,0 +1,187 @@
+    "Service </oppia-noninteractive-policy> and <oppia-noninteractive-policy link=\"privacy\">" +
+    "Privacy Policy </oppia-noninteractive-policy>."
+
+/** Tests for [PolicyPageTagHandler]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class PolicyPageTagHandlerTest {
+  @Rule
+  @JvmField
+  val mockitoRule: MockitoRule = MockitoJUnit.rule()
+
+  @Mock
+  lateinit var mockImageRetriever: FakeImageRetriever
+
+  @Mock
+  lateinit var mockPolicyPageLinkClickListener: PolicyPageLinkClickListener
+
+  @Captor
+  lateinit var policyPageCaptor: ArgumentCaptor<PolicyPage>
+
+  @Inject
+  lateinit var context: Context
+  @Inject lateinit var consoleLogger: ConsoleLogger
+
+  private lateinit var noTagHandlers: Map<String, CustomTagHandler>
+  private lateinit var tagHandlersWithPolicyPageSupport: Map<String, CustomTagHandler>
+  private lateinit var testView: TextView
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    noTagHandlers = mapOf()
+    tagHandlersWithPolicyPageSupport = mapOf(
+      CUSTOM_POLICY_PAGE_TAG to PolicyPageTagHandler(
+        mockPolicyPageLinkClickListener,
+        consoleLogger
+      )
+    )
+    testView = TextView(context)
+  }
+
+  // TODO(#3085): Introduce test for verifying that the error log scenario is logged correctly.
+
+  @Test
+  fun testParseHtml_emptyString_doesNotIncludeClickableSpan() {
+    val parsedHtml =
+      CustomHtmlContentHandler.fromHtml(
+        html = "",
+        imageRetriever = mockImageRetriever,
+        customTagHandlers = tagHandlersWithPolicyPageSupport
+      )
+
+    val clickableSpans = parsedHtml.getSpansFromWholeString(ClickableSpan::class)
+    assertThat(clickableSpans).isEmpty()
+  }
+
+  @Test
+  fun testParseHtml_withPolicyPageMarkup_includesClickableSpan() {
+    val parsedHtml =
+      CustomHtmlContentHandler.fromHtml(
+        html = POLICY_PAGE_LINK_MARKUP_1,
+        imageRetriever = mockImageRetriever,
+        customTagHandlers = tagHandlersWithPolicyPageSupport
+      )
+
+    val clickableSpans = parsedHtml.getSpansFromWholeString(ClickableSpan::class)
+    assertThat(clickableSpans).hasLength(2)
+  }
+
+  @Test
+  fun testParseHtml_withPolicyPageMarkup_clickSpan_callsClickListener() {
+    val parsedHtml =
+      CustomHtmlContentHandler.fromHtml(
+        html = POLICY_PAGE_LINK_MARKUP_1,
+        imageRetriever = mockImageRetriever,
+        customTagHandlers = tagHandlersWithPolicyPageSupport
+      )
+
+    val clickableSpans = parsedHtml.getSpansFromWholeString(ClickableSpan::class)
+    clickableSpans[0].onClick(testView)
+    clickableSpans[1].onClick(testView)
+    verify(mockPolicyPageLinkClickListener, times(2))
+      .onPolicyPageLinkClicked(capture(policyPageCaptor))
+
+    assertThat(policyPageCaptor.allValues).containsExactly(
+      PolicyPage.TERMS_OF_SERVICE,
+      PolicyPage.PRIVACY_POLICY
+    ).inOrder()
```

Comment:
Similarly to an earlier comment, suggest splitting this into two tests as distinct behaviors.

Reply:
Done.